src/libmongoc/src/mongoc/mongoc-secure-channel.c

```diff
@@ -929,7 +976,10 @@ mongoc_secure_channel_handshake_step_3 (mongoc_stream_tls_t *tls,
 
    /* check if the required context attributes are met */
    if (secure_channel->ret_flags != secure_channel->req_flags) {
-      MONGOC_ERROR ("Failed handshake");
+      MONGOC_LOG_AND_SET_ERROR (error,
+                                MONGOC_ERROR_STREAM,
+                                MONGOC_ERROR_STREAM_SOCKET,
+                                "Failed handshake");
```

Comment:
Optional: Similar comment. I suggest adding an error above L974 like the following:
```
MONGOC_LOG_AND_SET_ERROR (error,
                          MONGOC_ERROR_STREAM,
                          MONGOC_ERROR_STREAM_SOCKET,
                          "Invalid state: credentials unset");
```